addons/website_slides_survey/models/survey_survey.py

```diff
@@ -28,11 +28,12 @@ def _compute_slide_channel_data(self):
 
     @api.ondelete(at_uninstall=False)
     def _unlink_except_linked_to_course(self):
-        slides = self.slide_ids.filtered(lambda slide: slide.slide_type == "certification").exists()
-        if slides:
-            raise ValidationError(_("The Certification '%(certification_name)s' cannot be deleted because it is still being used by the Course(s) '%(courses_names)s'",
-                                    certification_name=self.title,
-                                    courses_names=', '.join(slides.mapped('channel_id.name'))))
+        for record in self:
+            slides = record.slide_ids.filtered(lambda slide: slide.slide_type == "certification").exists()
```

Comment:
The first implementation was better for performances due to the exists (better do it once).